.github/workflows/ci.yaml

```diff
@@ -15,7 +15,9 @@ jobs:
       with:
         name: pre-commit-hooks
         signingKey: '${{ secrets.CACHIX_SIGNING_KEY }}'
-    - run: nix-build --keep-going
+    - run: nix-build --keep-going --impure
+      env:
+        NIXPKGS_ALLOW_UNFREE: 1
```

Comment:
I'm not a huge fan of turning on this flag, but the Terraform version available in the current `nixpkgs` requires it. We could also try to pull an older version of Terraform like `1.5.7` before HashiCorp made the license change, but that would require setting up a new input like [nixpkgs-terraform](https://github.com/stackbuilders/nixpkgs-terraform).

Reply:
That is why I opted for `opentofu` instead of `terraform` in the `aws-vault` hook. I feel like most users would prefer using the Open Source fork, but I get what you are saying that the naming might be confusing.

I don't really use the hook myself and was just making changes to fix the CI build of `aws-vault` in https://github.com/cachix/devenv/pull/994, so I'm impartial to using one or the other.

However, I don't think that we should purposely use a pre-licence-change version of terraform in an separate input for the hook.

If users wish to use an old version that was released under a true Open Source, then I'd say they should just use a custom hook.

Reply:
Fair point; another option that I considered was to write a script that checks if the `tofu` executable is available and falls back to `terraform` if it isn't, something like this:

```nix
if [ -f ${hooks.terraform-format.package}/bin/tofu ]; then
  ${hooks.terraform-format.package}/bin/tofu fmt -check -diff "$@"
else
  ${hooks.terraform-format.package}/bin/terraform fmt -check -diff "$@"
fi
```

This way, we can keep the current implementation while making it easier for users to switch to Terraform if they want to. Part of the reason I submitted this PR in the first place is that it feels strange to use OpenToFu to look for formatting errors when using `languages.terraform.enable` on `devenv`; ideally, this hook should use the same version of OpenTofu or Terraform as `languages.opentofu.package` or `languages.terraform.package` respectively. The changes made in this PR would make it easier to mimic some of the behaviour found in the [languages.rust](https://github.com/cachix/devenv/blob/main/src/modules/languages/rust.nix#L92-L94) module within `devenv`, where the module passes the appropriate tooling to the `pre-commit` hooks to avoid installing multiple versions of the same package.